modules/spring-boot-autoconfigure/src/main/java/org/apache/ignite/springframework/boot/autoconfigure/IgniteAutoConfiguration.java

```diff
@@ -0,0 +1,101 @@
+ * limitations under the License.
+ */
+
+package org.apache.ignite.springframework.boot.autoconfigure;
+
+import org.apache.ignite.Ignite;
+import org.apache.ignite.Ignition;
+import org.apache.ignite.configuration.IgniteConfiguration;
+import org.springframework.beans.factory.BeanFactory;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnBean;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnMissingBean;
+import org.springframework.boot.context.properties.ConfigurationProperties;
+import org.springframework.boot.context.properties.EnableConfigurationProperties;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+
+/**
+ * Ignite node auto configuration.
+ * Spring will use this class to autoconfigure Ignite node on application startup.
+ *
+ * Initialization of the node will be the following:
+ * <ul>
+ *   <li>If {@link IgniteConfiguration} bean exists in {@link BeanFactory} it will be used for the node start.</li>
+ *   <li>If {@link IgniteConfiguration} bean doesn't exists following rules are applied:
+ *     <ol>
+ *       <li>If bean of the type {@link IgniteConfigurer} exists in {@link BeanFactory}
+ *         it will be used to customize empty {@link IgniteConfiguration} instance.
+ *         If you want to set custom SPI instances or similar hardcoded values.
+ *         You should do it in custom {@link IgniteConfigurer}.</li>
+ *       <li>Application properties will override config values. Prefix for properties names is "ignite".</li>
+ *     </ol>
+ *   </li>
+ * </ul>
+ *
+ * Example of application property:
+ * <pre>
+ *   ignite.igniteInstanceName=my-instance
+ * </pre>
+ *
+ * @see IgniteConfigurer
+ * @see IgniteConfiguration
+ * @see Ignite
+ */
+@Configuration
+@EnableConfigurationProperties
+public class IgniteAutoConfiguration {
+    /** Ignite configuration properties prefix. */
+    public static final String IGNITE_PROPS_PREFIX = "ignite";
+
+    /**
+     * @return Default(empty) Ignite configurer.
+     */
+    @ConditionalOnMissingBean
+    @Bean
+    public IgniteConfigurer nodeConfigurer() {
+        return cfg -> { /* No-op. */ };
+    }
+
+    /**
+     * Provides an instance of {@link IgniteConfiguration} customized with the {@link IgniteConfigurer}.
+     *
+     * @param configurer Configurer.
+     * @return Ignite configuration.
+     */
+    @ConditionalOnMissingBean
+    @Bean
+    @ConfigurationProperties(prefix = IGNITE_PROPS_PREFIX)
+    public IgniteConfiguration igniteConfiguration(IgniteConfigurer configurer) {
+        IgniteConfiguration cfg = new IgniteConfiguration();
+
+        configurer.accept(cfg);
+
+        return cfg;
+    }
+
+    /**
+     * Starts an {@link Ignite} node.
+     *
+     * @param cfg Configuration.
+     * @return Ignite instance.
+     */
+    @ConditionalOnBean(IgniteConfiguration.class)
+    @Bean
+    public Ignite ignite(IgniteConfiguration cfg) {
```

Comment:
I suppose it will the next iteration of the starter. This code isn't able to use more than one instance of Ignite.

Reply:
Can you, please, file a ticket?